ui/notification/toast.go

```diff
@@ -0,0 +1,77 @@
+package notification
+
+import (
+	"time"
+
+	"gioui.org/layout"
+	"gioui.org/op"
+
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+type (
+	C = layout.Context
+	D = layout.Dimensions
+)
+
+type Toast struct {
+	theme   *decredmaterial.Theme
+	success bool
+	message string
+	timer   *time.Timer
+
+	show bool
+}
+
+func NewToast(th *decredmaterial.Theme) *Toast {
+	return &Toast{
+		theme: th,
+	}
+}
+
+func (t *Toast) Notify(message string, success bool) {
+	t.message = message
+	t.success = success
+	t.timer = time.NewTimer(time.Second * 3)
+	t.show = true
+}
+
+func (t *Toast) Layout(gtx layout.Context) layout.Dimensions {
+	if !t.show {
+		return layout.Dimensions{}
+	}
+
+	t.handleToastDisplay(gtx)
```

Comment:
This check should be done at the top of the function.

Reply:
moving the handle function to the top of the Layout method would lead to a nil dereference error.